src/main/scala/akka/persistence/cassandra/journal/CassandraRecovery.scala

```diff
@@ -14,148 +18,54 @@ trait CassandraRecovery extends ActorLogging {
 
   implicit lazy val replayDispatcher = context.system.dispatchers.lookup(replayDispatcherId)
 
-  def asyncReplayMessages(persistenceId: String, fromSequenceNr: Long, toSequenceNr: Long, max: Long)(replayCallback: (PersistentRepr) => Unit): Future[Unit] = Future {
-    replayMessages(persistenceId, fromSequenceNr, toSequenceNr, max)(replayCallback)
-  }
-
-  def asyncReadHighestSequenceNr(persistenceId: String, fromSequenceNr: Long): Future[Long] = Future {
-    readHighestSequenceNr(persistenceId, fromSequenceNr)
-  }
-
-  def readHighestSequenceNr(persistenceId: String, fromSequenceNr: Long): Long = {
-    findHighestSequenceNr(persistenceId, math.max(fromSequenceNr, highestDeletedSequenceNumber(persistenceId)))
-  }
-
-  def readLowestSequenceNr(persistenceId: String, fromSequenceNr: Long): Long = {
-    new MessageIterator(persistenceId, fromSequenceNr, Long.MaxValue, Long.MaxValue).find(!_.deleted).map(_.sequenceNr).getOrElse(fromSequenceNr)
-  }
-
-  def replayMessages(persistenceId: String, fromSequenceNr: Long, toSequenceNr: Long, max: Long)(replayCallback: (PersistentRepr) => Unit): Unit = {
-    new MessageIterator(persistenceId, fromSequenceNr, toSequenceNr, max).foreach(msg => {
-      replayCallback(msg)
-    })
-  }
-
-  /**
-   * Iterator over messages, crossing partition boundaries.
-   */
-  class MessageIterator(persistenceId: String, fromSequenceNr: Long, toSequenceNr: Long, max: Long) extends Iterator[PersistentRepr] {
-
-    import PersistentRepr.Undefined
-
-    private val initialFromSequenceNr = math.max(highestDeletedSequenceNumber(persistenceId) + 1, fromSequenceNr)
-    log.debug("Starting message scan from {}", initialFromSequenceNr)
-
-    private val iter = new RowIterator(persistenceId, initialFromSequenceNr, toSequenceNr)
-    private var mcnt = 0L
+  private[this] implicit val mat = ActorMaterializer()(context.system)
 
-    private var c: PersistentRepr = null
-    private var n: PersistentRepr = PersistentRepr(Undefined)
+  private[this] val queries: CassandraReadJournal =
+    PersistenceQuery(context.system).readJournalFor[CassandraReadJournal](CassandraReadJournal.Identifier)
```

Comment:
hm, this is tricky. if the write side journal is **not** the default `cassandra-journal` and this will still point to the default `cassandra-query-journal`, which will point to `cassandra-journal`